src/main/java/appeng/fluids/client/gui/GuiFluidLevelEmitter.java

```diff
@@ -0,0 +1,216 @@
+
+	private GuiButton plus1;
+	private GuiButton plus10;
+	private GuiButton plus100;
+	private GuiButton plus1000;
+	private GuiButton minus1;
+	private GuiButton minus10;
+	private GuiButton minus100;
+	private GuiButton minus1000;
+
+	public GuiFluidLevelEmitter( final InventoryPlayer inventoryPlayer, final PartFluidLevelEmitter te )
+	{
+		super( new ContainerFluidLevelEmitter( inventoryPlayer, te ) );
+		this.levelEmitter = te;
+	}
+
+	@Override
+	public void initGui()
+	{
+		super.initGui();
+
+		this.level = new GuiNumberBox( this.fontRenderer, this.guiLeft + 24, this.guiTop + 43, 79, this.fontRenderer.FONT_HEIGHT, Long.class );
+		this.level.setEnableBackgroundDrawing( false );
+		this.level.setMaxStringLength( 16 );
+		this.level.setTextColor( 0xFFFFFF );
+		this.level.setVisible( true );
+		this.level.setFocused( true );
+		( (ContainerFluidLevelEmitter) this.inventorySlots ).setTextField( this.level );
+
+		final int y = 40;
+		final int x = 80 + 44;
+		this.guiSlots.add( new GuiFluidSlot( this.levelEmitter.getConfig(), 0, 0, x, y ) );
+	}
+
+	@Override
+	protected void addButtons()
+	{
+		this.redstoneMode = new GuiImgButton( this.guiLeft - 18, this.guiTop + 28, Settings.REDSTONE_EMITTER, RedstoneMode.LOW_SIGNAL );
+
+		final int a = AEConfig.instance().levelByMillyBuckets( 0 );
+		final int b = AEConfig.instance().levelByMillyBuckets( 1 );
+		final int c = AEConfig.instance().levelByMillyBuckets( 2 );
+		final int d = AEConfig.instance().levelByMillyBuckets( 3 );
+
+		this.buttonList.add( this.plus1 = new GuiButton( 0, this.guiLeft + 20, this.guiTop + 17, 22, 20, "+" + a ) );
+		this.buttonList.add( this.plus10 = new GuiButton( 0, this.guiLeft + 48, this.guiTop + 17, 28, 20, "+" + b ) );
+		this.buttonList.add( this.plus100 = new GuiButton( 0, this.guiLeft + 82, this.guiTop + 17, 32, 20, "+" + c ) );
+		this.buttonList.add( this.plus1000 = new GuiButton( 0, this.guiLeft + 120, this.guiTop + 17, 38, 20, "+" + d ) );
+
+		this.buttonList.add( this.minus1 = new GuiButton( 0, this.guiLeft + 20, this.guiTop + 59, 22, 20, "-" + a ) );
+		this.buttonList.add( this.minus10 = new GuiButton( 0, this.guiLeft + 48, this.guiTop + 59, 28, 20, "-" + b ) );
+		this.buttonList.add( this.minus100 = new GuiButton( 0, this.guiLeft + 82, this.guiTop + 59, 32, 20, "-" + c ) );
+		this.buttonList.add( this.minus1000 = new GuiButton( 0, this.guiLeft + 120, this.guiTop + 59, 38, 20, "-" + d ) );
+
+		this.buttonList.add( this.redstoneMode );
+	}
+
+	@Override
+	public void drawBG( final int offsetX, final int offsetY, final int mouseX, final int mouseY )
+	{
+		super.drawBG( offsetX, offsetY, mouseX, mouseY );
+		this.level.drawTextBox();
+	}
+
+	@Override
+	protected boolean drawUpgrades()
+	{
+		return false;
+	}
+
+	@Override
+	protected String getBackground()
+	{
+		return "guis/lvlemitter.png";
+	}
+
+	@Override
+	protected GuiText getName()
+	{
+		return GuiText.FluidLevelEmitter;
+	}
+
+	@Override
+	protected void handleButtonVisibility()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1186)